mobile-browser-based-version/src/logic/training/round_tracker.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Class that keeps track of everything round related.
+ *
+ * @remark
+ * A round is unit of time measured in epochs. The client wishes to share his weights with the
+ * server every r rounds. If r = 1.5 that means every 1.5 epochs, the onBatchEnd(batch, log) callback function
+ * of TF .fit is used to know when a round is done. Since the given batch in onBatchEnd is not cumulative, i.e.
+ * it is 0 after an epoch is done, we need to keep track of the current epoch in training to compute the
+ * currentBatchNumberSinceStart which is required in order to know if a round is done.
+ *
+ */
+export class RoundTracker {
+    round: number;
+    epoch: number;
+    roundDuration: number;
+    numberOfBatchesInAnEpoch: number
+
+    constructor (roundDuration: number, trainSize: number, batchSize: number, epoch: number = 0) {
+      this.round = 0
+      this.epoch = epoch
+      this.roundDuration = roundDuration
+      this.numberOfBatchesInAnEpoch = RoundTracker.numberOfBatchesInAnEpoch(trainSize, batchSize)
+
+      console.log(`RoundTracker: roundDuration: ${roundDuration}, nb batches in an epoch ${this.numberOfBatchesInAnEpoch}`)
+    }
+
+    /**
+   * Returns true if a local round has ended, false otherwise.
+   *
+   * @remark
+   * Batch is the current batch, this goes from 1, ... , batchSize*trainSize.
```

Comment:
typo here, batch size doesn't matter, but you meant trainsize/batchsize. anyway would be nicer to just let batch count to infinity, not reset to zero every epoch